opentracing-impl/src/main/java/io/opentracing/propagation/TextMapImpl.java

```diff
@@ -16,7 +16,7 @@
 import java.util.Iterator;
 import java.util.Map;
 
-public final class TextMapImpl implements TextMapWriter, TextMapReader {
+public final class TextMapImpl implements TextMap {
```

Comment:
In other languages we haven't provided text map implementations as part of the core API. Not that it's needed by `-impl`, either: the only caller of TextMapImpl is test code at the moment.

As for splitting it into reader/writer pieces: I can see it either way. If we're going to provide helpers for the `Map<String, String>` case, though, I think we can make the intention clearer through naming. I'll give it a shot.